verticapy/tests_new/pipeline/test_helper.py

```diff
@@ -0,0 +1,172 @@
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+import pytest
+
+from verticapy._utils._sql._sys import _executeSQL
+
+from verticapy.pipeline import parser, _helper
+
+import verticapy.sql.sys as sys
+
+
+def test_required_keywords():
```

Comment:
Maybe a more cleaner way to do this test is using parameterization. Advantages:
1- More accurate test errors
2- Cleaner and easier to interpret

Also you should separate the negative and positive tests.

So something like this:
```
@pytest.mark.parametrize(
    "yaml, keywords",
    [
        ({"key1": 1, "key2": 2, "key3": 3}, ["key1", "key2", "key3"]),
        (...)
    ]
)
def test_required_keywords(yaml,keywords):
    """
    test function required_keywords.
    """
    yaml = {"key1": 1, "key2": 2, "key3": 3}
    keywords = ["key1", "key2", "key3"]
    assert _helper.required_keywords(yaml, keywords) == True

    yaml = {}
    keywords = []
    assert _helper.required_keywords(yaml, keywords) == True

    yaml = {"key1": 1}
    keywords = []
    assert _helper.required_keywords(yaml, keywords) == True


@pytest.mark.parametrize(
    "yaml, keywords","error_keyword".
    [
        ({"key1": 1, "key3": 3}, ["key1", "key2", "key3"],"key22"),
        (...)
    ]
)
def test_required_keywords_negative(yaml,keywords,error_keyword):
    """
    negative test for function required_keywords.
    """

    with pytest.raises(KeyError) as error_info:
        _helper.required_keywords(yaml, keywords)
    assert error_keyword in str(error_info.value)
```